src/System.Management.Automation/utils/PathUtils.cs

```diff
@@ -366,18 +366,25 @@ internal static DirectoryInfo CreateModuleDirectory(PSCmdlet cmdlet, string modu
 
                 if (string.IsNullOrEmpty(rootedPath))
                 {
-                    string personalModuleRoot = ModuleIntrinsics.GetPersonalModulePath();
-                    if (string.IsNullOrEmpty(personalModuleRoot))
+                    if (Path.IsPathRooted(moduleNameOrPath))
```

Comment:
If the requirement is for an absolute path and moduleNameOrPath could already satisfy this, shouldn't this check be on line 360 to check this first before all the other attempts?

Reply:
The `ResolveRootedFilePath` call on line 360 handles some special cases when the path is rooted, such as whether the drive is a PowerShell PSDrive, and whether the rooted path contains wildcards. Directly returning before calling it will likely break something.

Do the check here is the safest, because before adding the `ThrowTerminatingError`, `Path.Combine` would just return `path2` when `Path.IsPathRooted(path2)` returns `true`. See the code here: https://source.dot.net/#System.Private.CoreLib/Path.cs,643

Reply:
I don't understand why `moduleNameOrPath` can be rooted after `ResolveRootedFilePath` returns null (unable to resolve rooted path).  Please add a comment that demonstrates when this can happen and why this check is needed.

Reply:
`GetResolvedProviderPathFromPSPath` throws when the path doesn't exist. I agree that it may not be the right method to be used here. (maybe `GetUnresolvedProviderPathFromPSPath`?)

In this scenario, the passed-in rooted path is a directory in temp folder that doesn't exist yet, and `directoryInfo.Create()` will be called in this case on line 410 of `CreateModuleDirectory`.

The method `ResolveRootedFilePath` has 15 references, so changing `GetResolvedProviderPathFromPSPath` to `GetUnresolvedProviderPathFromPSPath` might break something. If we want, we should look into this separately.
For this fix, we want to backport it to 7.0.x and 7.2.x, so it should be as safe as possible.

Reply:
Some comments were added.